server.py

```diff
@@ -29,11 +30,10 @@ def handle_client(client_socket):
 # Todo: Modify the message to contain more info (timestamp, node_id)
 def send_notifications(other_nodes):
     for node in other_nodes:
-        port = int(other_nodes[node][1])
         with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as node_socket:
-            node_socket.connect(('localhost', port))
+            node_socket.connect(node)
```

Comment:
Saan tästä virheen: `TypeError: connect(): AF_INET address must be tuple, not str`.  
Pitäisi varmaan olla muodossa `node_socket.connect((<ip>,<port>))`.